test/test_filesystem.py

```diff
@@ -0,0 +1,97 @@
+# -*- coding: utf-8 -*-
+import os.path
+import picard.formats
+import unittest
+import shutil
+from contextlib import suppress
+
+from PyQt5 import QtCore
+from picard import config
+from tempfile import mkdtemp
+from test.test_formats import FakeTagger
+
+
+settings = {
+    'enabled_plugins': '',
+    'move_files': True,
+    'move_additional_files': True,
+    'move_additional_files_pattern': 'cover.jpg',
+}
+
+
+class TestFileSystem(unittest.TestCase):
+
+    def setUp(self):
+        self.src_directory = mkdtemp()
+        self.tgt_directory = mkdtemp()
+        QtCore.QObject.tagger = FakeTagger()
+        config.setting = settings.copy()
+
+    def tearDown(self):
+        shutil.rmtree(self.src_directory)
+        shutil.rmtree(self.tgt_directory)
+        pass
+
+    def _set_up_src_file(self, filename, src_rel_path):
+        """Copy filename to the src directory under src_rel_path."""
+        path = os.path.join(self.src_directory, src_rel_path)
+        shutil.copy(filename, path)
+        return path
+
+    def _set_up_tgt_filename(self, tgt_rel_path):
+        """Return the absolute path to tgt_rel_path in the tgt directory."""
+        return os.path.join(self.tgt_directory, tgt_rel_path)
+
+    def _prepare_files(self, src_rel_path='', tgt_rel_path=''):
+        """Prepare src files and tgt filenames for a test."""
+        with suppress(FileExistsError):
+            os.mkdir(os.path.join(self.src_directory, src_rel_path))
+
+        # Prepare the source directory structure under self.src_directory
+        # .../<src_rel_path>/test.mp3
+        # .../<src_rel_path>/cover.jpg
+
+        old_filename = \
+            self._set_up_src_file(os.path.join('test', 'data', 'test.mp3'),
+                                  os.path.join(src_rel_path, 'test.mp3'))
+        old_additional_filename = \
```

Comment:
We are not that strict on line lengths, even 120-130 chars is fine. Please avoid using `\`. It makes for some very ugly to read code.

Reply:
ok